src/Microsoft.IdentityModel.TestExtentions/TestTokenCreator.cs

```diff
@@ -0,0 +1,399 @@
+            Modulus = Base64UrlEncoder.DecodeBytes("vfPmqjbDDsT_xnh5CfnR97GirbRASP3rKyUlLOpytrkkWtNSD2S0ovnFlW5H4E9OiGusUpeJYD6HmAY5VSafWKC0qVEAvFIT_Fs6j8kJ3kiGsJc9Q-jktds06nvUMn1I-qxmLI7flw0khko6LCi6nTrBrPQ_XlQLYGS0C6-wkyCn0qqB6zsvX8kEElE7bEka0RkWzBVIWsVKkllajBNEeUVvAmy4KvmbFxXDlleNDXsJcETyo3-GmpaevrAr7ryIOOZuUQfkA8Hd-rb-lVkOkk2M-SChNlW8zNeyCksjp0LOF6r8SDf0fgnsKWsQ7qN2_Ltuvmd1X_HMIfdOmhi5Y6W6WnzWacO-jqbTb1k37bASbj2ViYB5-puTfEABzc12KN2mLRTA2TqhmYhy-9EPeDWdXsJwXKP1gtBrxXGFyEzuD4Y8lRYetY0RI5IJc8SoeUauJu75kNxUp-ffteIqP-uhG-E-BcFxIj_YfmI8Yui_XHU7OwDHTBdXk-ahBE7l-kvI-KN2hFcO1DIYK-c2no7fv1CXW8Lz-w7XWl54jn2MyKx4PnTUYT8langL6ZW3xux2wLrnbGwnIisWNWh0ih2ggKiiLCiOxJM9bwA-VbFSyihb3vWs-i4RAQtUwkvV6-8nyZv-SMcI9B7LuziJohnp_gIJZ8FIvtjoJHfGrZk"),
+            P = Base64UrlEncoder.DecodeBytes("z3zVMYG1_TWwnNs99hyMJI9xs1zrwbV_STv9cqpaBxgPIHqpDyoG68a_VZF2RfuYLKT9ksEfLmgDeBPuM_vwBXSUApRqQo46WOpYUaDTcQJUJcspYQENQDkvtMGTuZQd6wrfGksDkUgnuZZ2CAKjN6vzGGlUkLJ9eEihbVTsCcYpp1sSK_J5HZCu0MZcnEU4oMbWKRHHApkQC8TJkutRw1XjY6s_ABj7M2W2D9CIO-CnUaXxmh21JNaJ2zP0nK1cfWgzvIikVoey7FKbGbOoIBnVf9AZUAVfclBLWI3wfUUVod_1lwLA07-7kvQ6SCEiHpsep3RwONcYJJH3vD3WlQ"),
+            Q = Base64UrlEncoder.DecodeBytes("6l2EkIkJDFV_djyTFdpYbjrmg1lIRa1_oFF1925M0aDAEeJncRYkBOyupE9Ni6vHSTlRGN7I3Vs6V72Jnt_wpKqxz0qdhUseO89_JaqQE65euYq1QyeAjnHg64q4PcLYmBDub9Hn6y0xkTzkMCOUe5ZNGeosb5LW4qQVGyXnyeg9PmwFrBjQwr_Na6-1VrS2HCqzxvQWkOuJdjt3XB4ODXUnze9G-8nx8d2Cb9hC2qT4QxTPCq25i0n2DvIcRs_pD6RyNPg7EYOSH2Alc1vVzj7aYW9Plx69rIrtIFxH0ZwHsHua-AjDg7IZLK_Ghc3RV-9nTPEA2_6q_A_oOvfN9Q"),
+        };
+
+    /// <summary>
+    /// Gets or sets the issuer to be stamped on the tokens created.
+    /// </summary>
+    public string Issuer { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Audience to be stamped on the tokens created.
+        /// </summary>
+        public string Audience { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SigningCredentials used to sign the tokens created.
+        /// </summary>
+        public SigningCredentials SigningCredentials { get; set; }
+
+        #region Create Test Token Methods
+        /// <summary>
+        /// Creates a default valid test token based based on the class's <see cref="Issuer"/>, <see cref="Audience"/>
+        /// and <see cref="SigningCredentials"/> values.
+        /// </summary>
+        /// <returns>A test JWS token which is valid according to the class configuration.</returns>
+        public string CreateDefaultValidToken()
+        {
+            var tokenDescriptor = CreateTokenDescriptorWithInstanceOverrides();
+            return CreateToken(tokenDescriptor);
+        }
+
+        /// <summary>
+        /// Creates a test token with a signature that doesn't match the payload.
+        /// </summary>
+        /// <returns>A test JWS token with a signature which doesn't match the payload.</returns>
```

Comment:
```suggestion
        /// <returns>A test JWS token with a signature which doesn't match the payload.</returns>
        /// <example>
        ///  The following code shows how the class under test handles a token with a malformed signature.
        /// <code>
        /// [Fact]
        /// public void TokenWithBadSignature()
        /// {
        ///     TestWithGeneratedToken(
        ///         testTokenCreator.CreateTokenWithInvalidSignature,
        ///         typeof(ArgumentException),
        ///         "IDX14111");
        ///  }
        /// </code></example>
```

This enables developers to see how to use the API from the reference documentation.
cc: @mmacy

Reply:
Hmm this example assumes folks are using Xunit and abstracts away the actual test code (i.e. this is just a wrapper that doesn't give a great idea of the actual logic they'd want to implement.)

I think the overall suggestion is a good one. Let me think about how best to share code examples that would be most useful.